source/MaterialXTest/MaterialXRenderGlsl/RenderGlsl.cpp

```diff
@@ -712,12 +712,15 @@ void  GlslShaderRenderTester::runBake(mx::DocumentPtr doc, const mx::FileSearchP
     {
         bakeHeight = 2;
     }
+    const mx::FilePath bakedImagepath = outputFileName.getParentPath();
+    bakedImagepath.createDirectory();
     mx::Image::BaseType baseType = bakeHdr ? mx::Image::BaseType::FLOAT : mx::Image::BaseType::UINT8;
     mx::TextureBakerPtr baker = mx::TextureBaker::create(bakeWidth, bakeHeight, baseType);
     baker->setupUnitSystem(doc);
     baker->setTargetUnitSpace("meter");
     baker->setImageHandler(_renderer->getImageHandler());
-    baker->setOutputResourcePath(outputFileName.getParentPath());
+    baker->setOutputResourcePath(bakedImagepath);
+    baker->setOptimizeConstants(true);
```

Comment:
should this be configurable instead of always on?

Reply:
Eventually yes should be driven via _options.mltx